sources/HUBViewController.m

```diff
@@ -1174,8 +1175,10 @@ - (void)headerAndOverlayComponentViewsWillAppear
 - (void)componentWrapperWillAppear:(HUBComponentWrapper *)componentWrapper
 {
     [componentWrapper viewWillAppear];
-    
-    if (componentWrapper.isContentOffsetObserver) {
+
+    BOOL wasContentOffsetUpdated = !self.lastContentOffset || !CGPointEqualToPoint([self.lastContentOffset CGPointValue], self.collectionView.contentOffset);
```

Comment:
Sure. Basically otherwise you're relying on implicit integer conversion between a pointer and `nil`, which means that you are technically comparing across different types (it only so happens to work because `nil` gets implicitly converted to `0`, both as a pointer and as a literal). This is why Swift prohibits these type of comparisons, as it has a much more static and thorough type system.